src/Microsoft.Extensions.OData.Migration/MigrationExtension.cs

```diff
@@ -0,0 +1,148 @@
+﻿// ------------------------------------------------------------------------------
+// <copyright company="Microsoft Corporation">
+//     Copyright © Microsoft Corporation. All rights reserved.
+// </copyright>
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Extensions.OData.Migration
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Xml;
+    using Microsoft.AspNetCore.Builder;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.AspNetCore.Routing;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.OData.Migration.Filters;
+    using Microsoft.Extensions.OData.Migration.Formatters.Deserialization;
+    using Microsoft.Extensions.OData.Migration.Formatters.Serialization;
+    using Microsoft.OData;
+
+    /// <summary>
+    /// Contains extension method for IApplicationBuilder to use translation middleware provided by this project.
+    /// </summary>
+    public static class MigrationExtension
+    {
+        /// <summary>
+        /// Extension method to use V3 to V4 translation middleware.
+        /// </summary>
+        /// <param name="builder">IApplicationBuilder that will use translation middleware</param>
+        /// <param name="v3Edmx">V3 edmx to send back when requested for metadata</param>
+        /// <param name="v4Model">Required V4 model to validate translated URI</param>
+        /// <returns>builder now using migration middleware</returns>
+        public static IApplicationBuilder UseODataMigration(this IApplicationBuilder builder,
```

Comment:
> IApplicationBuilder UseODataMigration(this IApplicationBuilder builder, [](start = 21, length = 72)

how can customer use the whole extensions methods to build his service? Would you please create a ASP.NET CORE application sample to play with them?

Reply:
The extension is being used in Intune's StatelessConnectorService: https://msazure.visualstudio.com/One/_git/Intune-Svc-RACerts?path=%2Fsrc%2FServices%2FConnectorService%2FSource%2FStartup.cs&version=GBdevelop&line=144&lineStyle=plain&lineEnd=144&lineStartColumn=13&lineEndColumn=79

I will make sure it's doc'd clearly in the md file.